From 3fbe86302749fe4496dde699714b3876e706af63 Mon Sep 17 00:00:00 2001
From: kboeluek <kboeluek@90b65887-3827-0410-9a23-83215b262276>
Date: Wed, 14 May 2014 17:26:52 +0000
Subject: [PATCH 0465/1000] MANTIS 0020990 Mantis 0020990 extension;
 Platform_Extension_CUSS_1_2: Periodic/Automatic restart of the application

- added interface definitions for indicationType(enum) and data(struct)

git-svn-id: svn://localhost/SelfServiceCommon/trunk@674 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/idl/Massai/appcontrol.idl          | 19 ++++++++++++++++---
 1 file changed, 16 insertions(+), 3 deletions(-)

diff --git a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
index 9bc63895..2ead3e65 100644
--- a/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
+++ b/SelfServiceCommon/Massai/idl/Massai/appcontrol.idl
@@ -1,5 +1,4 @@
 // basic definition for the massai application control service  (internal)
-// M_INTERFACE_VERSION="1.0.0.0" 
 // History:
 // V1.0.0.0
 // application is deprecated; retained for compatibility with old MCC
@@ -23,6 +22,8 @@
 
 module appControl
 {
+    const string interface_version = "1.1";
+   
     // list of strings
     typedef sequence<string> strings;
 
@@ -56,7 +57,6 @@ module appControl
     /** Defines the application data that is passed by the ApplicationManager to this device
      *  @note The cmdLine must be given as EXECUTABLE.EXE params: param1 param2 ...
      */
-
     struct applicationStart
     {
       string appName;     /**< The application name as internally handeled by the ApplicationManager*/
@@ -85,6 +85,19 @@ module appControl
 
       resolutions currentResolution();
     };
+
+    enum indicationType
+    {
+      IND_RESOURCE_USAGE,
+      IND_UNDEFINED
+    };
+    
+    struct data
+    {
+      string applicationName;
+      string applicationMemoryUsage;
+      string applicationTimeUsage;
+    };
 };
 
-#endif // APPCONTROL_IDL
\ No newline at end of file
+#endif // APPCONTROL_IDL
-- 
2.41.0.windows.1

